geopy/geocoders/mapbox.py

```diff
@@ -0,0 +1,209 @@
+            return Location(location, (latitude, longitude), place)
+        if exactly_one:
+            return parse_feature(features[0])
+        else:
+            return [parse_feature(feature) for feature in features]
+
+    @staticmethod
+    def _coerce_point_to_string(point):
+        """
+        Do the right thing on "point" input. For geocoders with reverse
+        methods.
+        """
+        if isinstance(point, Point):
+            return ",".join((str(point.longitude), str(point.latitude)))
+        elif isinstance(point, (list, tuple)):
+            return ",".join((str(point[1]), str(point[2])))  # -altitude
+        elif isinstance(point, string_compare):
+            return ",".join(point.split(",")[::-1])
+        else:
+            raise ValueError("Invalid point")
+
+    def geocode(
+            self,
+            query,
+            exactly_one=True,
+            proximity=None,
+            country=None,
+            bbox=None,
+            timeout=DEFAULT_SENTINEL
+    ):
+        """
+        Return a location point by address
+
+        :param str query: The address or query you wish to geocode.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param proximity: a coordinate to bias local results based on a provided location.
+        :type proximity: list of tuple
+
+        :param string country: Country to filter result in form of
+            ISO 3166-1 alpha-2 country code.
+
+        :param bbox: The bounding box of the viewport within which
+            to bias geocode results more prominently.
+        :type bbox: list or tuple
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+        """
+        params = {}
+
+        params['access_token'] = self.api_key
+
+        if query is None or query == '':
+            raise GeocoderQueryError('query must be set')
+        if bbox:
+            if len(bbox) != 4:
+                raise GeocoderQueryError(
+                    "bbox must be a four-item iterable of lat,lon,lat,lon"
+                )
+            params['bbox'] = self._format_bbox_param(bbox)
+
+        if country:
+            if len(country) != 2:
+                raise GeocoderQueryError(
+                    "country must be specified as ISO 3166-1 alpha-2 country code."
+                )
+            params['country'] = country
+
+        if proximity:
+            if len(proximity) != 2:
+                raise GeocoderQueryError(
+                    "proximity must be a two-item iterable of lat,lon"
+                )
+            params['proximity'] = self._format_proximity_param(proximity)
+
+        request_url = self.api + "/" + quote(query) + ".json/"
```

Comment:
I believe it should be `quote(query.encode('utf-8'))`.

This concatenation might look better (IMO) if we do the following:
1. Introduce class attribute: `geocode_path = '/geocoding/v5/mapbox.places/%(query)s.json/'`
2. In `__init__` do: `self.api = "%s://%s%s" % (self.scheme, self.domain, self.geocode_path)`
3. Here do the following:

        quoted_query = quote(query.encode('utf-8'))
        url = "?".join((self.api % dict(query=quoted_query),
                        urlencode(params)))

You may find an example of this approach here: https://github.com/geopy/geopy/blob/0a3a60bd6f9cb570e346ffef7953babf630ae8a3/geopy/geocoders/tomtom.py#L117

It's just a suggestion. This concatenation might be fine as it is (except the missing `.encode('utf-8')`).